### ISSUE TITLE
Document how to use the Gradle plugin with the Kotlin DSL

### ISSUE THREAD

**Issue by jmfayard** [square/wire#1992]
I had troubles setting up the plugin with the Kotlin DSL

I wanted to contribute the setup I had found to work, but I couuldn't edit the mkdocs website, so here it is:

### settings.gradle.kts

```kotlin
pluginManagement {
    repositories {
        gradlePluginPortal()
        mavenCentral()
    }
    resolutionStrategy {
        eachPlugin {
            if (requested.id.id == "com.squareup.wire") {
                useModule("com.squareup.wire:wire-gradle-plugin:<version>")
            }
        }
    }
}
```

### build.gradle.kts

```kotlin
plugins {
    id("com.squareup.wire")
}

wire {
    kotlin {
    }
}

```

**Comment by oldergod** [comment]
Why do you need the resolution strategy?

**Comment by jmfayard** [comment]
@oldergod it doesn't work for me without the resolutionStrategy
See https://github.com/LouisCAD/kotlin-libraries-playground/commit/cc97b611e0a22f833690f6ebff917db2d0efeaa5

**Comment by rossdanderson** [comment]
I had the same issue, thank you for including the working example

**Comment by boswelja** [comment]
Is this still an issue with 4.0? If so it would be great to get a fix rather than applying a resolution strategy workaround

**Comment by stanislavnastasiu-dev** [comment]
> Is this still an issue with 4.0? If so it would be great to get a fix rather than applying a resolution strategy workaround

unfortunately that is still the case

**Comment by boswelja** [comment]
> > Is this still an issue with 4.0? If so it would be great to get a fix rather than applying a resolution strategy workaround
> 
> unfortunately that is still the case

Fortunately it doesn't actually appear to be. You should be able to define `id 'com.squareup.wire' version '4.9.1'` with no extra workarounds

**Comment by stanislavnastasiu-dev** [comment]
> > > Is this still an issue with 4.0? If so it would be great to get a fix rather than applying a resolution strategy workaround
> > 
> > 
> > unfortunately that is still the case
> 
> Fortunately it doesn't actually appear to be. You should be able to define `id 'com.squareup.wire' version '4.9.1'` with no extra workarounds

you're right! i got the same error and when i saw the issue still being open i tought that was still the workaround to be used. Turns out all i needed was to add 
`apply false` in the project gradle
